src/services/movieService.ts

```diff
@@ -0,0 +1,96 @@
+import { Prisma } from "@prisma/client";
+import prisma from "../../prisma";
+
+interface MovieFilter {
+  title?: string;
+  genreId?: number;
+}
+
+interface MovieSort {
+  sortBy?: "title" | "releaseDate";
+  sortOrder?: "asc" | "desc";
+}
+
+export const listMovies = async (
+  page: number = 1,
+  pageSize: number = 10,
+  filterBy: MovieFilter = {},
+  sortBy: MovieSort
+) => {
```

Comment:
I think it would be nice to add a response interface.

Reply:
Thank you for the suggestion. I will consider adding a response interface to improve clarity and consistency.